jwst/resample/resample_step.py

```diff
@@ -170,12 +171,20 @@ def _check_list_pars(vals, name, min_vals=None):
             raise ValueError(f"Both '{name}' values must be either None or not None.")
 
     @staticmethod
-    def _load_custom_wcs(asdf_wcs_file, output_shape):
-        if not asdf_wcs_file:
+    def _load_custom_wcs(input_wcs_file, output_shape):
+        if not input_wcs_file:
             return None
 
-        with asdf.open(asdf_wcs_file) as af:
-            wcs = deepcopy(af.tree["wcs"])
+        try:
+            # File is ASDF file with a top-level wcs in the tree
+            with asdf.open(input_wcs_file) as af:
+                wcs = deepcopy(af.tree["wcs"])
+        except (KeyError, ValueError):
+            # File is ImageModel or SlitModel with the wcs under meta
```

Comment:
Can this be a `SlitModel` in a `ResampleStep`?

Reply:
Theoretically yes, but practically I think it will always be `ImageModel`.

Reply:
This function is also called by `ResampleSpecStep`, right?
Does passing an external WCS to spectral observations work?

Reply:
> This function is also called by `ResampleSpecStep`, right? Does passing an external WCS to spectral observations work?

It should.